### ISSUE TITLE
Azure Native 2.0 uses preview API for Iot Hub not supported in all regions

### ISSUE THREAD

**Issue by ProgrammerAl** [pulumi/pulumi-azure-native#2585]
### What happened?

After updating to Azure Native 2.0, I'm unable to create a new Azure Iot Hub resource in the East US region. Looks like the generated `Pulumi.AzureNative.Devices.IotHubResource` C# class has a comment saying it was generated from a preview API. That preview version is only available in a few Azure regions, and is not recommended to be used for production scenarios.

```csharp
    /// <summary>
    /// The description of the IoT hub.
    /// Azure REST API version: 2022-11-15-preview. Prior API version in Azure Native 1.x: 2020-08-31
    /// </summary>
    [AzureNativeResourceType("azure-native:devices:IotHubResource")]
    public partial class IotHubResource : global::Pulumi.CustomResource
```

As a result, the error message I get when running `pulumi up` is:
`azure-native:devices:IotHubResource iot-hub **creating failed** error: Code="NoRegisteredProviderFound" Message="No registered resource provider found for location 'eastus' and API version '2022-11-15-preview' for type 'IotHubs'. The supported api-versions are '2015-08-15-preview, 2016-02-03, 2017-01-19, 2017-07-01, 2018-01-22, 2018-04-01, 2018-04-01-preview, 2018-12-01-preview, 2019-03-22-preview, 2019-03-22, 2019-07-01-preview, 2019-11-04, 2020-01-01, 2020-03-01, 2020-04-01, 2020-06-15, 2020-07-10-preview, 2020-08-01, 2020-08-31-preview, 2020-08-31, 2021-02-01-preview, 2021-03-03-preview, 2021-03-31, 2021-07-01-preview, 2021-07-01, 2021-07-02-preview, 2021-07-02, 2022-04-30-preview'. The supported locations are 'westus, australiacentral2, australiasoutheast, canadaeast, centralus, eastus, germanywestcentral, koreasouth, japanwest, southindia, southafricawest, uaenorth, southeastasia, westeurope, norwaywest, westus3, norwayeast, uksouth, southcentralus, australiacentral, australiaeast, brazilsouth, canadacentral, centralindia, eastasia, francecentral, germanynorth, japaneast, koreacentral, northeurope, southafricanorth, uaecentral, ukwest, westus2, westcentralus, northcentralus, eastus2, qatarcentral, switzerlandnorth'."`

### Expected Behavior

As of this writing, it should probably be version 2021-07-02. https://learn.microsoft.com/en-us/azure/templates/microsoft.devices/iothub-allversions

### Steps to reproduce

1. With a C# Pulumi application, add a reference to the Pulumi.AzureNative version 2.0.0 NuGet.
2. Create a new instance of Pulumi.AzureNative.Devices.IotHubResource in the EastUS region.

### Output of `pulumi about`

CLI
Version      3.75.0
Go Version   go1.20.5
Go Compiler  gc

Host
OS       Microsoft Windows 11 Home
Version  10.0.22621 Build 22621
Arch     x86_64

This project is written in dotnet: executable='C:\Program Files\dotnet\dotnet.exe' version='7.0.306'

Backend
Name           pulumi.com
URL            https://app.pulumi.com/al-purple-spike
User           al-purple-spike
Organizations  al-purple-spike, Purple-Spike

### Additional context

Before upgrading to version 2.0 I was using the 2022-04-30-preview API version. It also looks like that version isn't available anymore. So I can't go back to that version.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Hi @ProgrammerAl, thank you for reporting this! We'll look into fixing this soon.

**Comment by thomas11** [comment]
Hi @ProgrammerAl, we do indeed use lots of preview versions. The problem is that some Azure services use preview versions quite liberally and their latest stable is very old. We curated the versions we used in v2 of the provider but in your case we got it wrong.

I just tried [the IotHubResource example from our docs](https://www.pulumi.com/registry/packages/azure-native/api-docs/devices/iothubresource/) but imported `devices/v20210702` instead of the default version. That's the latest stable version in [their docs](https://learn.microsoft.com/en-us/azure/templates/microsoft.devices/iothub-allversions).

I had to comment the `rootCertificate` and `ipVersion` properties which don't exist in the stable version, then I could run `pulumi up`. It got past "No registered resource provider found", but failed with

> Requested IoT Hub features '\"MinimumTlsVersion1_2, DataResidency\"' not available in 'eastus' region

Now I'm confused as to what parts of Microsoft.Devices are available where.
